<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>publication</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./images/logo.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-layout-custom">
<!-- sidebar -->
<!-- margin-sidebar -->
<!-- main -->





    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.6.0/css/all.min.css" integrity="sha512-Kc323vGBEqzTmouAECnVceyQqyqdsSiqLQISBL29aUW4U/M7pSPA/gEUZQqv1cwx4OnYxTxve5UMg5GT6L4JJg==" crossorigin="anonymous" referrerpolicy="no-referrer">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet">
  <link href="https://getbootstrap.com/docs/5.2/assets/css/docs.css" rel="stylesheet">
  <link href="https://unpkg.com/boxicons@2.1.1/css/boxicons.min.css" rel="stylesheet">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://cssslider.com/sliders/demo-16/engine1/style.css">
    <link rel="stylesheet" href="http://cssslider.com/sliders/demo-16/engine1/ie.css">
      <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD" crossorigin="anonymous">
      <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css">

    <script type="text/javascript" src="http://cssslider.com/sliders/demo-16/engine1/ie.js"></script>
    <script type="text/javascript" src="https://cssslider.com/sliders/demo-16/engine1/gestures.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js"></script>
    <script src="https://unpkg.com/@popperjs/core@2.11.6/dist/umd/popper.min.js"></script>
<script src="https://unpkg.com/tippy.js@6.3.3/dist/tippy-bundle.umd.min.js"></script>
<script src="https://unpkg.com/@popperjs/core@2.11.6/dist/umd/popper.min.js"></script>
<script src="https://unpkg.com/tippy.js@6.3.3/dist/tippy-bundle.umd.min.js"></script>
<link href="https://stackpath.bootstrapcdn.com/bootstrap/5.3.0/css/bootstrap.min.css" rel="stylesheet">
<script src="https://stackpath.bootstrapcdn.com/bootstrap/5.3.0/js/bootstrap.bundle.min.js"></script>
<script src="https://cdn.jsdelivr.net/gh/bwva/HTML-SpinnerElem/src/spinnerComponent.js"></script>



  <div class="flex items-center justify-center w-full"><!--<div class="center"></div>--><nav class="navbar navbar-expand fixed-top py-md-0" id="navbarcolor">
      <div class="container-fluid">
        <a class="navbar-brand" href="#"><img src="./images/logo.png">
        </a>
<!--        <button class="navbar-toggler p-4" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span>
        </button>-->
<!--        <div class="collapse navbar-collapse" id="navbarSupportedContent">
-->          <ul class="navbar-nav mx-auto"><li class="nav-item active"><a class="nav-link" href="./index.html">Home</a></li><li class="nav-item"><a class="nav-link" href="./instrument.html">Molecular BioLab</a></li><li class="nav-item"><a class="nav-link" href="./program.html">Computational Biology</a></li><li class="nav-item"><a class="nav-link" href="https://hprovax.github.io/DAiR2024.github.io/">Workshops</a></li><li class="nav-item"><a class="nav-link" href="./teach.html">Teaching</a></li><li class="nav-item"><a class="nav-link" href="./publication.html">Publication</a></li><li class="nav-item"><a class="nav-link" href="./about.html">About</a></li><!--<li class="nav-item"><a class="nav-link" href="contact.qmd">Contact</a></li>-->
          </ul>
<!--          <form class="form-inline"><div class="form-group"><input class="form-control" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-light"  type="submit" style="  justify-content: center;
 ">Submit</button></div>
          </form>-->
          <!--<input class=".checkbox_humb" type=".checkbox">-->
          <!--<div class="hamburger-lines">
              <span class="line line1"></span>
              <span class="line line2"></span>
              <span class="line line3"></span>
          </div>  -->
          <!--</div>--></div>
    </nav><div class="blog-featured-image"><div class="my_pic_background"><div class="my_pic"><img src="images/itsme.jpeg" alt="Hamed Abdollahi"></div><div class="info my_pic_text">
            <h1>Hamed Abdollahi</h1>
            <h2>Molecular Biologist</h2>
            <h3 id="card_subhead_I">Genomics Postdoctoral Fellow</h3>
            <h3 id="card_subhead_II">DVM, PhD</h3>
          </div><div class="my_pic_social"><div class="social-icons"><a href="mailto:hprovax@gmail.com" class="social-icon">
          <i class="fa-solid fa-envelopes-bulk"></i>
        </a>
        <!--2nd-->
        <a href="mailto:HA25@mailbox.sc.edu" class="social-icon ">
            <i class="fa-sharp fa-solid fa-building-columns"></i>
          </a>        
        <!--3nd-->
        <a href="https://www.linkedin.com/in/hamed-abdollahi-15117079" class="social-icon">
            <i class="fa-brands fa-linkedin"></i>
          </a>
          <!--4rd-->
          <a href="https://github.com/HProVax" class="social-icon">
            <i class="fa fa-github"></i>
          </a>
          <!--5th-->
          <a href="https://scholar.google.com/citations?user=lXf0_H0AAAAJ&amp;hl=en" class="social-icon">
            <i class="fa-brands fa-google-scholar"></i> 
          </a>
          <!--6th-->
          <a href="https://orcid.org/0000-0002-2202-2193" class="social-icon">
            <i class="fa-brands fa-orcid"></i> 
            
          </a>
          <!--7th-->
          <a href="https://maps.app.goo.gl/yTz8M8XeHsyMa3xk6" class="social-icon">
            <i class="fa-solid fa-earth-americas"></i> 
          </a></div></div></div></div>
    <div class="task-topics">
        <div class="wrapper_inst">
            <div class="pubrow">
                <div class="col-md-12">
                    <div class="page-content-html-pub">
      <div class="pubcol-lg-9 ">
        <script>
    function expand_abstract(input_value){
      var input_value = input_value;
      var obj=document.getElementById(input_value);
      if(obj.style.display=="block"){
        obj.style.display="none"
        }else{
          obj.style.display="block"
          
        }
      
    }
    </script>
    <div class="mb-8">
      <!--item1-2025>
            <div class="paper tool highlight llm">
        <h5>For More Publications, Visit My Google Scholar Profile</h5>
        <ol>
          <!-- <div class="myDIV"> -->
          <div class="row">
            <div class="col-5">
              <img src="images/rdc.jpg" class="card-img" id="pub_image">
            </div>
            <div class="col-7">
              <span class="text-black">
                <span class="text-primary">
                  <strong>[Current Opinion in Structural Biology 2023]</strong><br>
                </span> Computational modeling multiple conformational states of proteins with residual dipolar coupling data</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Hamed Abdollahi*, James H. Prestegard, Homayoun Valafar <br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('RDC2023');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://www.sciencedirect.com/science/article/pii/S0959440X2300129X?via%3Dihub" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="RDC2023" class="myhide">
                                                    <span class="text-500">Solution nuclear magnetic resonance spectroscopy provides unique opportunities to study the structure and dynamics of biomolecules in aqueous environments. While spin relaxation methods are well recognized for their ability to probe timescales of motion, residual dipolar couplings (RDCs) provide access to amplitudes and directions of motion, characteristics that are important to the function of these molecules. Although observed in the 1960s, the acquisition and computational analysis of RDCs has gained significant momentum in recent years, and particularly applications to motion in proteins have become more numerous. This trend may well continue as RDCs can easily leverage structures produced by new computational methods (e.g., AlphaFold) to produce functional descriptions. In this report, we provide examples and a summary of the ways that RDCs have been used to confirm the existence of internal dynamics, characterize the type of dynamics, and recover atomic-scale structural ensembles that define the full range of conformational sampling.<br></span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
      <!--item1-->
      <div class="paper tool highlight llm">
        <h5>For More Publications, Visit My Google Scholar Profile</h5>
        <ol>
          <!-- <div class="myDIV"> -->
          <div class="row">
            <div class="col-5">
              <img src="images/rdc.jpg" class="card-img" id="pub_image">
            </div>
            <div class="col-7">
              <span class="text-black">
                <span class="text-primary">
                  <strong>[Current Opinion in Structural Biology 2023]</strong><br>
                </span> Computational modeling multiple conformational states of proteins with residual dipolar coupling data</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Hamed Abdollahi*, James H. Prestegard, Homayoun Valafar <br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('RDC2023');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://www.sciencedirect.com/science/article/pii/S0959440X2300129X?via%3Dihub" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="RDC2023" class="myhide">
                                                    <span class="text-500">Solution nuclear magnetic resonance spectroscopy provides unique opportunities to study the structure and dynamics of biomolecules in aqueous environments. While spin relaxation methods are well recognized for their ability to probe timescales of motion, residual dipolar couplings (RDCs) provide access to amplitudes and directions of motion, characteristics that are important to the function of these molecules. Although observed in the 1960s, the acquisition and computational analysis of RDCs has gained significant momentum in recent years, and particularly applications to motion in proteins have become more numerous. This trend may well continue as RDCs can easily leverage structures produced by new computational methods (e.g., AlphaFold) to produce functional descriptions. In this report, we provide examples and a summary of the ways that RDCs have been used to confirm the existence of internal dynamics, characterize the type of dynamics, and recover atomic-scale structural ensembles that define the full range of conformational sampling.<br></span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
                                <!-- This will be your separation line -->
                                <!--item2-->
      <div class="paper alg llm highlight">
        <ol>
          <div class="row">
            <div class="col-5">
              <img src="images/DGE.png" class="card-img" id="pub_image">
            </div>
            <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[NISBRE 2024]</strong><br>
                                                    </span> GENE EXPRESSION PROFILES UNDER THE INFLUENCE OF KYNURENIC ACID DURING PROGENY</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Hamed Abdollahi, Courtney Wright, Ana Pocivavsek, Homayoun Valafar<br> 
                                                    </span>
                                                    [<a href="javascript:expand_abstract('NISBRE2024');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://nisbre2024.vfairs.com/en/abstract-page/#abstract"><span class="text-primary">Paper</span></a>]
                                                </div>
                                            </div>
                                        <div id="NISBRE2024" class="myhide">
                                            <span class="text-500">Tryptophan metabolism plays a crucial role in sleep disorders and is also associated with various neurodegenerative diseases. Kynurenic acid, a metabolite of tryptophan, exhibits uneven distribution in different parts of rat brains. Increased production of kynurenic acid due to sleep disorders leads to its accumulation in various brain sectors. The effects of kynurenic acid have been studied during fetal, pre, and post-natal development in rats. Since genes play a significant role in sleep regulation, this study aims to identify
differentially expressed genes between male and female rats exposed to increased
concentration of kynurenic acid during the fetal period.Among 2060 significant over expressed genes, 199 genes exhibited differential expression patterns in females across various time points, while 216 genes displayed distinct expression changes in males.</span>
                                        </div>  
                                    </ol>
                                </div>
                                     <div class="separator"></div>                                <!-- This will be your separation line -->
                                <!--item3-->
      <div class="paper alg llm highlight">
        <ol>
          <div class="row">
            <div class="col-5">
              <img src="images/h52021.jpg" class="card-img" id="pub_image">
            </div>
            <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[Infection, Genetics and Evolution 2020]</strong><br>
                                                    </span> Spatial spread and emergence of reassortant H5 highly pathogenic avian influenza viruses in Iran</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Hamed Abdollahi, Ali Safar Maken Ali, Isabella Monne, Adelaide Milani, Alice Fusaro, Bianca Zecchin<br> 
                                                    </span>
                                                    [<a href="javascript:expand_abstract('h52020');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://www.sciencedirect.com/science/article/pii/S1567134820301738#f0005" target="_blank"><span class="text-primary">Paper</span></a>]
                                                </div>
                                            </div>
                                        <div id="h52020" class="myhide">
                                            <span class="text-500">Since 2005, H5Nx highly pathogenic avian influenza (HPAI) viruses of the Goose/Guangdong (Gs/GD) lineage have spread worldwide, affecting poultry and wild birds in Asia, Europe, Africa and North America. So far, the role of Western Asia and the Middle East in the diffusion dynamics of this virus has been poorly explored. In order to investigate the genetic diversity and the role of Iran in the transmission dynamics of the Gs/GD lineage, we sequenced the complete genome of twenty-eight H5Nx viruses which were circulating in the country between 2016 and 2018. We reported the first characterization of the HPAI H5N6 subtype of clade 2.3.4.4B in Iran and gave evidence of the high propensity of the Gs/GD H5 AIVs to reassort, describing six novel H5N8 genotypes of clade 2.3.4.4B, some of them likely generated in this area, and one H5N1 reassortant virus of clade 2.3.2.1c. Our spatial analyses demonstrated that the viruses resulted from different viral introductions from Asia and Europe and provided evidence of virus spread from Iran to the Middle East. Therefore, Iran may represent a hot-spot for virus introduction, dissemination and for the generation of new genetic variability. Increasing surveillance efforts in this high-risk area is of utmost importance for the early detection of novel emerging strains with zoonotic potential.</span>
                                        </div>  
                                    </ol>
                                </div>
                                     <div class="separator"></div>

      <!--item3-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="images/ibvpro.png" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[Arch Virol 2021]</strong></span><br> Coronavirus: proteomics analysis of chicken kidney tissue infected with variant 2 (IS-1494)-like avian infectious bronchitis virus</span> 
                                                    <br>
                                                    <span class="text-500">
                                                            Hamed Abdollahi, Mostafa Rezaei-Tavirani, Arash Ghalyanchilangeroudi, Hossein Maghsoudloo, Masood Hashemzadeh, Hossein Hosseini &amp; Abbas Barin<br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('ibvpro');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://link.springer.com/article/10.1007/s00705-020-04845-7#author-information" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="ibvpro" class="myhide">
                                                    <span class="text-500">Avian infectious bronchitis virus is one of the most important gammacoronaviruses, which causes a highly contagious disease. In this study, we investigated changes in the proteome of kidney tissue of specific-pathogen-free (SPF) chickens that were infected with an isolate of the nephrotropic variant 2 genotype (IS/1494/06) of avian coronavirus. Twenty 1-day-old SPF White Leghorn chickens were randomly divided into two groups, each comprising 10 chickens, which were kept in separate positive-pressure isolators. Chickens in group A served as a virus-free control group up to the end of the experiment, whereas chickens in group B were inoculated with 0.1 ml of 104.5 EID50 of the IBV/chicken/Iran/UTIVO-C/2014 isolate of IBV, and kidney tissue samples were collected at 2 and 7 days post-inoculation (dpi) from both groups. Sequencing of five protein spots at 2 dpi and 22 spots at 7 dpi that showed differential expression by two-dimensional electrophoresis (2DE) along with fold change greater than 2 was done by MS-MALDI/TOF/TOF. Furthermore, the corresponding protein-protein interaction (PPI) networks at 2 and 7 dpi were identified to develop a detailed understanding of the mechanism of molecular pathogenesis. Topological graph analysis of this undirected PPI network revealed the effect of 10 genes in the 2 dpi PPI network and nine genes in the 7 dpi PPI network during virus pathogenesis. Proteins that were found by 2DE analysis and MS/TOF-TOF mass spectrometry to be down- or upregulated were subjected to PPI network analysis to identify interactions with other cellular components. The results show that cellular metabolism was altered due to viral infection. Additionally, multifunctional heat shock proteins with a significant role in host cell survival may be employed circuitously by the virus to reach its target. The data from this study suggest that the process of pathogenesis that occurs during avian coronavirus infection involves the regulation of vital cellular processes and the gradual disruption of critical cellular functions.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>                                                                <div class="separator"></div>

                                <!--item4-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="images/AImodels.png" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[Cancers 2024]</strong></span><br>  Integrating Omics Data and AI for Cancer Diagnosis and Prognosis </span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Yousaku Ozaki, Phil Broughton, Hamed Abdollahi, Homayoun Valafar, and Anna V. Blenda  <br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('aimodels');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://www.mdpi.com/2072-6694/16/13/2448" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="aimodels" class="myhide">
                                                    <span class="text-500">Cancer is one of the leading causes of death, making timely diagnosis and prognosis very important. Utilization of AI (artificial intelligence) enables providers to organize and process patient data in a way that can lead to better overall outcomes. This review paper aims to look at the varying uses of AI for diagnosis and prognosis and clinical utility. PubMed and EBSCO databases were utilized for finding publications from 1 January 2020 to 22 December 2023. Articles were collected using key search terms such as “artificial intelligence” and “machine learning.” Included in the collection were studies of the application of AI in determining cancer diagnosis and prognosis using multi-omics data, radiomics, pathomics, and clinical and laboratory data. The resulting 89 studies were categorized into eight sections based on the type of data utilized and then further subdivided into two subsections focusing on cancer diagnosis and prognosis, respectively. Eight studies integrated more than one form of omics, namely genomics, transcriptomics, epigenomics, and proteomics. Incorporating AI into cancer diagnosis and prognosis alongside omics and clinical data represents a significant advancement. Given the considerable potential of AI in this domain, ongoing prospective studies are essential to enhance algorithm interpretability and to ensure safe clinical integration.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
    

<!--item5-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="./images/heal2024.jpg" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[IJBM2024]</strong></span><br> Polydopamine contained hydrogel nanocomposites with combined antimicrobial and antioxidant properties for accelerated wound healing</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Mahin Abdollahi, Sina Andalib, Roghayeh Ghorbani, Davoud Afshar, Mohammad Gholinejad, Hamed Abdollahi, Ali Akbari, Nasser Nikfarjam<br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('heal2024');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://www.sciencedirect.com/science/article/pii/S0141813024025054?via%3Dihub" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="heal2024" class="myhide">
                                                    <span class="text-500">Overproduction of reactive oxygen species (ROS) in infected wounds induces a tremendous inflammatory reaction to delay wound healing. To address this problem, we designed a multifunctional polyacrylamide/PVA-based hydrogel containing synthesized poly(1-glycidyl-3-butylimidazolium salicylate) (polyGBImSal) and fabricated polydopamine-coated polyphenolic nanosheet (PDA@PNS) for wound dressing. The PDA@PNS particles were designed to induce I) antioxidant and anti-inflammatory features through ROS-scavenging and II) cell adhesive properties by the existing polydopamine into the hydrogels. The poly(ionic liquid)-based polyGBImSal was designed to allocate effective hydrogel antimicrobial activity. The fabricated hydrogel nanocomposites showed excellent properties in the swelling ratio, cell adhesiveness, protein adsorption, and anti-inflammatory, proving their general performance for application in wound healing. Furthermore, these hydrogels showed high antimicrobial activity (over 95 %) against three common wound-infecting pathogenic microbes: Escherichia coli, Staphylococcus aureus, and Candida albicans. The healing process of full-thickness dermal wounds in rats was accelerated by applying hydrogel nanocomposites with 0.5 wt% of PDA@PNS and 28 wt% of polyGBImSal. The wound closure contraction attained full closure, reaching 100 %, after 14 days, contrasted with the control group employing commercial wound dressing (Tegaderm), which achieved a closure rate of 68 % within the equivalent timeframe. These results make these hydrogel nanocomposites promising candidates for multifunctional wound dressing applications.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
    
<!--item6-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="images/is1494.png" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[VirusDisease2018]</strong></span><br> Complete genome analysis of Iranian IS-1494 like avian infectious bronchitis virus</span> 
                                                    <br>
                                                    <span class="text-500">
                                                            Fatemeh Sadat Mousavi, Arash Ghalyanchilangeroudi, Hossein Hosseini, Bahar Nayeri Fasaei, Seyed Ali Ghafouri, Hamed Abdollahi, Mohammad Hosein Fallah-Mehrabadi &amp; Naser Sadri<br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('gpt4aigchip');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://link.springer.com/article/10.1007/s13337-018-0462-4" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="gpt4aigchip" class="myhide">
                                                    <span class="text-500">The nephropathogenic infectious bronchitis virus (IBV) strain IS-1494 like (variant-2; GI-23) was first isolated in the Middle East (1998). Despite intensive vaccinations, IS-1494 like IBVs are still circulating in Iran (the dominant genotype) and spread to other countries. Here, the full-length genome of this Iranian IS-1494 like IBV was (Mahed) determined to understand its evolutionary relationships. The genome consists of 27,652 nucleotides, with mutations in most of the structural genes. Thirteen open reading frames (ORFs) were predicted in the Mahed isolate (5′ UTR-1a-1b-S-3a-3b-E-M-4b-4c-5a-5b-N-6b-3′ UTR). ORFs 4b, 4c, and 6b, which has rarely been reported, were present in the Mahed genome. According to phylogenetic analysis of the full-length genome, 1a, S2, M, E, N protein, Mahed isolate clustered with the QX type strain. Based on the partial 1b, S1, Mahed clustered with the Q1 strain. The full-length genome of Mahed isolate shared the highest sequence homology with Gray and JMK (90.06–90.07%) and was least related to the Vic-s (86.21%). These data show that evolutionary variation because of recombination in IBV plays a major role in the adaptation and origin of IBV leading to new genetic and types of the virus strain.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
    
<!--item7-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="images/h52017.png" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[Tropical Animal HP 2017]</strong></span><br> Clade 2.3.4.4 avian influenza A (H5N8) outbreak in commercial poultry</span> 
                                                    <br>
                                                    <span class="text-500">
                                                            Seyed Ali Ghafouri, Arash GhalyanchiLangeroudi, Hossein Maghsoudloo, Reza KH Farahani, Hamed Abdollahi, Farshad Tehrani &amp; Mohammad Hossein Fallah<br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('h52017');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://link.springer.com/article/10.1007/s11250-017-1302-z" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="h52017" class="myhide">
                                                    <span class="text-500">In 2010, H5N8 highly pathogenic avian influenza (HPAI) viruses of the A/Goose/Guangdong/1/1996 lineage dramatically affected poultry and wild birds in Asia, Europe, and North America. In November 2016, HPAI H5N8 was detected in a commercial layer farm in Tehran province. The diagnosis was based on real-time reverse transcriptase PCR (RRT-PCR) and sequencing of haemaglutinin (HA) and neuraminidase (NA) genes from suspected samples. Genetic and phylogenetic analysis of the HA gene demonstrated that the Iranian HPAI H5N8 viruses belong to the HPAI H5 virus clade 2.3.4.4 and cluster within group B (Gochang-like). In particular, the highest similarity was found with the sequences of the HPAI H5N8 identified in Russia in 2016. To our knowledge, this clade has not been previously detected in Iran. Previous HPAI A (H5) epidemic in Iran occurred in 2015 and involved exclusively viruses of clade 2.3.2.1c. These findings indicate that Iran is at high risk of introduction of HPAI H5 of the A/Goose/Guangdong/1/1996 lineage from East Asia and highlight the need to maintain adequate monitoring activities in target wild and domestic bird species for HPAI early detection. This study is useful for better understanding the genetic and antigenic evolution of H5 HPAI viruses in the region and the world.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
    

<!--item8-->
      <div class="paper tool highlight llm">
                                    
                                    <ol>
                                        <!-- <div class="myDIV"> -->
                                            
                                            <div class="row">
                                                <div class="col-5">
                                                    <img src="images/transonet.png" class="card-img" id="pub_image">
                                                </div>
                                                <div class="col-7">
                                                    <span class="text-black"><span class="text-primary">
                                                        <strong>[CSCI 2023]</strong></span><br> TransONet: Automatic Segmentation of Vasculature in Computed Tomographic Angiograms Using Deep Learning</span> 
                                                    <br>
                                                    <span class="text-500">
                                                        Alireza Bagheri Rajeoni, Breanna Pederson, Ali Firooz, Hamed Abdollahi, Andrew K. Smith, Daniel G. Clair, Susan M. Lessner, Homayoun Valafar<br>
                                                    </span>
                                                    [<a href="javascript:expand_abstract('CSCI2023');"><span class="text-primary">Abstract</span></a>]
                                                    [<a href="https://arxiv.org/abs/2311.10328" target="_blank"><span class="text-primary">Paper</span></a>]<br>
                                                </div>
                                                <div id="CSCI2023" class="myhide">
                                                    <span class="text-500">Pathological alterations in the human vascular
system underlie many chronic diseases, such as atherosclerosis
and aneurysms. However, manually analyzing diagnostic images
of the vascular system, such as computed tomographic angiograms
(CTAs) is a time-consuming and tedious process. To address this
issue, we propose a deep learning model to segment the vascular
system in CTA images of patients undergoing surgery for
peripheral arterial disease (PAD). Our study focused on
accurately segmenting the vascular system (1) from the descending
thoracic aorta to the iliac bifurcation and (2) from the descending
thoracic aorta to the knees in CTA images using deep learning
techniques. Our approach achieved average Dice accuracies of
93.5% and 80.64% in test dataset for (1) and (2), respectively,
highlighting its high accuracy and potential clinical utility. These
findings demonstrate the use of deep learning techniques as a
valuable tool for medical professionals to analyze the health of the
vascular system efficiently and accurately. Please visit the GitHub
page for this paper at https://github.com/pip-alireza/TransOnet.<br> </span>
                                                </div>
                                            </div>
                                        <!-- </div>  -->
                                    
                                        <!-- <div class="myhide">
                                            
                                        </div>   -->
                                   </ol>

                                </div>
                                <div class="separator"></div>
    
          </div>
          
      </div>
      </div>
          </div>

    </div>
<!--End-->
                            





<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</div></div></div></body></html>
